sky/skylet/providers/gcp/config.py

```diff
@@ -805,7 +814,15 @@ def get_usable_vpc(config) -> str:
         if len(vpcnets_all) == 1:
             # Skip checking any firewall rules if the user has specified a VPC.
             logger.info(f"Using user-specified VPC {specific_vpc_to_use!r}.")
-            return specific_vpc_to_use
+            subnets = _list_subnets(
```

Comment:
Nit: Doesn't feel very clean calling _list_subnets multiple times.

Is it possible to use the pattern in AWS's config.py? Call _list_subnets() once to get all `[(subnet, vpc, ...)]` and directly sort + filter this list.

Reply:
Good point! Just changed the implementation to list subnets first, which will save at most #VPC times gcp API call.

Tested:
- [x] `sky launch --cpus 2 --cloud gcp`
- [x] `sky launch --gpus L4 --cloud gcp` (default VPC does not have the first region to be tried for L4 GPU, i.e. us-east-4, and it will correctly use skypilot-vpc instead)